compiler/luci-interpreter/src/core/Tensor.cpp

```diff
@@ -68,10 +53,6 @@ void Tensor::writeData(const void *data_ptr, size_t data_size)
   std::memcpy(data<void>(), data_ptr, data_size);
 }
 
-void Tensor::resize(const Shape &new_shape)
```

Comment:
Q) is data allocated at this point?
Maybe add an assert?

Reply:
As I understand, at this method data can be not allocated. And what assert can be good at this method?

Reply:
I meant something like this:
``` c++
assert(!_data_allocated);
```

this is a proposal, not requirement (I think we need to check this locally first :slightly_smiling_face: )

Reply:
Q. What happens if at this point data is allocated and new shape is more than old? I affraid that user of `Tensor` may expect that tensor contain buffer with size `shape.num_elements() * sizeof(datatype)`  but tensor not has enought space.

Reply:
But sometimes we can call this method, when data already allocated. For example in kernel Concatenation test. This assert will fail at this test.

> I meant something like this:
> 
> ```c++
> assert(!_data_allocated);
> ```
> 
> this is a proposal, not requirement (I think we need to check this locally first 🙂 )

Reply:
In this situation user should call allocate_memory method. And after that old buffer will release and then memory manager will allocate new buffer for the new shape.

> Q. What happens if at this point data is allocated and new shape is more than old? I affraid that user of `Tensor` may expect that tensor contain buffer with size `shape.num_elements() * sizeof(datatype)` but tensor not has enought space.

Reply:
(optional) It would be nice to have comment with statement why this method is safe =)